core/system/src/mpu/vmpu_freescale_k64_mem.c

```diff
@@ -60,17 +126,13 @@ void vmpu_mem_switch(uint8_t src_box, uint8_t dst_box)
 
     box = &g_mem_box[dst_box];
 
-    /* for box zero, just disable private ACL's */
-    src_count = g_mem_box[src_box].count;
-    if(src_box && !dst_box)
-    {
-        /* disable private regions */
-        if(src_count)
-        {
-            t = g_mem_acl_user + 1;
-            while(src_count--)
-                MPU->WORD[t++][3] = 0;
-        }
+    /* Disable all user MPU regions. */
+    for (t = K64F_MPU_REGIONS_USER; t < K64F_MPU_REGIONS_MAX; t++) {
+        MPU->WORD[t][3] = 0;
+    }
+
+    /* for box zero, just return. */
+    if(src_box && !dst_box) {
```

Comment:
@niklas-arm : you only need to disable the MPU regions that remain after copying all new regions in. You can ensure that by checking the position of the mpu index after all other stuff has been done and zero everything from that point up to zero. I am pretty sure we did that for v7M